src/cpp/hu.bme.mit.inf.modes3.ESP32-sensor/railsensor_3/ESP32-sensor/lib/SensorMonitor/sensor_monitor.cpp

```diff
@@ -0,0 +1,230 @@
+
+#include "sensor_monitor.h"
+
+
+float convertRawGyro(int gRaw) {
+  // since we are using 250 degrees/seconds range
+  // -250 maps to a raw value of -32768
+  // +250 maps to a raw value of 32767
+
+  float g = (gRaw * 250.0) / 32768.0;
+
+  return g;
+}
+
+
+void updateCallback(void *arg) {
+  //Serial.println("collecting color and gyro data...");
+  SensorMonitor *moni = static_cast<SensorMonitor *>(arg);
+  while(1) {
+    moni->update();
+    vTaskDelay(1);
+  }
+}
+
+
+void printHex(byte *buffer, byte bufferSize) {
+  for (byte i = 0; i < bufferSize; i++) {
+    //Serial.print(buffer[i] < 0x10 ? " 0" : " ");
```

Comment:
Consider using some sort of logging mechanism instead of commenting out the `Serial.print` commands.

Reply:
And why are you iterating over the buffer, when you are clearly not doing anything?